### PR TITLE
Enable Snowflake to use Spark for extract/load

### PR DESCRIPTION
**Pull request by hsubbaraj-spiral**
## Describe your changes and why you are making these changes
Enable Snowflake to use Spark for extract/load.
## Related issue number (if any)
Eng-2298
## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/python/aqueduct_executor/operators/connectors/data/snowflake.py [hunk 1]
```diff
@@ -1,11 +1,50 @@
+from typing import Any, Callable, Dict, List, Optional
+
 from aqueduct_executor.operators.connectors.data import config, relational, utils
+from aqueduct_executor.operators.connectors.data import connector, extract, load
+from aqueduct_executor.operators.utils.enums import ArtifactType
 from sqlalchemy import create_engine, engine
+from pyspark.sql import SparkSession, DataFrame
```
Comment (saurav-c) [inline src/python/aqueduct_executor/operators/connectors/data/snowflake.py:7]: Is this backwards compatible with current snowflake integrations in use? Or will it break if `pyspark` is not installed?

File: src/python/aqueduct_executor/operators/connectors/data/snowflake.py [hunk 2]
```diff
@@ -20,3 +59,19 @@ def _create_engine(config: config.SnowflakeConfig) -> engine.Engine:
...
     return create_engine(url)
+
+
+def _map_to_snowflake_mode(params: load.RelationalParams) -> str:
+    """
+    Map Aqueduct update modes to Snowflake update modes found here:
+    https://docs.snowflake.com/ko/developer-guide/snowpark/reference/python/api/snowflake.snowpark.DataFrameWriter.save_as_table.html
+    """
+    snowflake_update_mode = "ignore"
+    if params.update_mode.value == "append":
```
Comment (saurav-c) [inline src/python/aqueduct_executor/operators/connectors/data/snowflake.py:70]: replace these with the enums defined [here](https://github.com/aqueducthq/aqueduct/blob/main/src/python/aqueduct_executor/operators/connectors/data/common.py#L23)

**Review by likawind — COMMENTED** [review]
LGTM! I will defer to @saurav-c for the final call

File: src/python/aqueduct_executor/operators/connectors/data/snowflake.py [hunk 3]
```diff
@@ -4,11 +4,11 @@
 
 class SnowflakeConnector(relational.RelationalConnector):
     def __init__(self, config: config.SnowflakeConfig):
-        conn_engine = _create_engine(config)
+        conn_engine = create_snowflake_engine(config)
         super().__init__(conn_engine)
 
 
-def _create_engine(config: config.SnowflakeConfig) -> engine.Engine:
+def create_snowflake_engine(config: config.SnowflakeConfig) -> engine.Engine:
```
Comment (saurav-c) [inline src/python/aqueduct_executor/operators/connectors/data/snowflake.py:11]: Doesn't look like this is imported anywhere. Is there a reason for the name change?

**Review by saurav-c — APPROVED** [review]
Left a nit and reminder to check linter

**Pull request merged**